### PR TITLE
[front] Remove the logic of upserting the `document_id` as `parents[0]` from front

### PR DESCRIPTION
**Pull request by aubin-tchoi**
## Description

- This PR removes the logic of upserting the `document_id` as `parents[0]` in front.
- The idea is to leave it up to connectors to set the parents, only doing validation in front (we already have logs/metrics).

## Risk

- anything that breaks here is something to change in connectors.

## Deploy Plan

- Deploy front.

### CODE REVIEW

File: front/lib/api/data_sources.ts [hunk 1]
```diff
@@ -382,14 +382,6 @@ export async function upsertDocument({
   const documentId = name;
   const documentParents = parents || [];
 
-  // Ensure that the documentId is included in the parents as the first item.
-  // remove it if it's already present and add it as the first item.
-  const indexOfDocumentId = documentParents.indexOf(documentId);
-  if (indexOfDocumentId !== -1) {
-    documentParents.splice(indexOfDocumentId, 1);
-  }
-  documentParents.unshift(documentId);
```
Comment (aubin-tchoi) [inline front/lib/api/data_sources.ts:391]: should we return an `Err` to crash more directly?

Comment (tdraier) [reply front/lib/api/data_sources.ts:391]: We know we have cases where we get into this (first entry in parents is not the document id) - especially if documentId is not in the same format as parents , so we should not crash until we're sure we sending parents[] with same format as documentId

Comment (tdraier) [reply front/lib/api/data_sources.ts:391]: IMO we should just not do this, or maybe only if parents is null or empty array. @Fraggle I think you added this , do you remember in which case it was used ?

Comment (spolu) [reply front/lib/api/data_sources.ts:391]: Yes extremely hard to understand a system that mutates what you give it. The system should check / enforce not modify

Comment (spolu) [reply front/lib/api/data_sources.ts:391]: Now the question is who wrote that and are they relying on it in any way?

**Review by spolu — APPROVED** [review]
LGTM

Need to check with the author if they are relying on it in any way.

File: front/lib/api/data_sources.ts (hunk 1 above)
Comment (Fraggle) [reply front/lib/api/data_sources.ts:391]: I was adding it to ensure that parent was the first id, you can remove the code until it's fully expected.

Comment (Fraggle) [reply front/lib/api/data_sources.ts:391]: BTW, we had a [discussion](https://github.com/dust-tt/dust/pull/8716#discussion_r1846621472), this method is not called by connectors (unless i'm mistaken), only by front when we are uploading to folders.

Comment (tdraier) [reply front/lib/api/data_sources.ts:391]: right, not this one, but the handleDataSourceTableCSVUpsert below is used by connectors upsert (through v1 api) . We should also remove the code there

Comment (aubin-tchoi) [reply front/lib/api/data_sources.ts]: As per half-IRL half-discussion here: we keep these 2 for now but remove in `handleDataSourceTableCSVUpsert`

**Pull request merged**